File: iOS/HelloManager.m


#import <Foundation/Foundation.h>

#import "RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(HelloManager, NSObject)

RCT_EXTERN_METHOD(sayHello:(NSString *)name
                  //callback:(RCTResponseSenderBlock *)successCallback)
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: iOS/stitchchat/HelloManager.swift


import Foundation

@objc(HelloManager)
class HelloManager: NSObject {
  
  @objc func sayHello(name: String,
                      resolver resolve: RCTPromiseResolveBlock,
                      rejecter reject : RCTPromiseRejectBlock) -> Void {
    let output = name + "Ram"
    /*let resultsDict = [
                      "success"  : true,
                      "contents" : output
                      ];*/
    // Date is ready to use!
    //return output
    resolve(output)
  }
  
}